lighthouse-cli/random-port.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import {createServer} from 'http';
+
+/**
+ * Return a random, unused port.
+ */
+function getRandomPort(): Promise<number> {
+  return new Promise((resolve, reject) => {
```

Comment:
I'm still fuzzy on when we're using `new Promise((resolve, reject)` vs. `Promise.resolve().then(()...`. You use the latter above. Drop some knowledge!